### DISCUSSION TITLE
Lookup服务：使用 .Result 异步方法获取值，会导致死锁问题。

### DISCUSSION THREAD

**Discussion by SmartCode-X** [dotnetcore/BootstrapBlazor#4805] [Q&A]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe the problem.

public class LookupServiceBaseEntity(ISysUserService sysUserService) : LookupServiceBase
{
    private ISysUserService SysUserService { get; } = sysUserService;

    public override IEnumerable<SelectedItem>? GetItemsByKey(string? key, object? data)
    {
        IEnumerable<SelectedItem>? items = null;
        if (key == "Foo.Complete")
        {
            var users = SysUserService.GetAllUsersAsync().Result;
            //var localizer = Provider.GetRequiredService<IStringLocalizer<Foo>>();//多语言支持
            items = users.Select(user => new SelectedItem
            { Value = user.Id.ToString(), Text = user.Name }).ToList();

        }
        return items;
    }
}
![1](https://github.com/user-attachments/assets/799ae3d0-5df1-43a0-a91d-470f352bd3da)

### Describe the solution you'd like

可以 使用 .Result 异步方法获取值。

**Comment by bb-auto** [comment]
@SmartCode-X Thank you for reporting. We will give triage later.

**Comment by BrandonStudio** [comment]
`.Result`是同步方法啊，阻塞不是正常现象吗

**Comment by ArgoZhang** [comment]
@SmartCode-X @BrandonStudio Blazor 的世界里可以说有两条定理

```
async void
```

```
.Result
```

大概率都会遇到问题。各种各样的问题。

**Comment by BrandonStudio** [comment]
@SmartCode-X 请尝试以下解决方案：
1. **使用 `Task.Run`**
   ```csharp
   var users = Task.Run(async () => await SysUserService.GetAllUsersAsync()).GetAwaiter().GetResult();
   ```
2. **把 `GetAllUsersAsync` 中的所有异步调用等待都改成 `continueOnCapturedContext: false`**
   例如
   ```diff
   - var x = await FooAsync(param1);
   + var x = await FooAsync(param1).ConfigureAwait(false);
   ```
   参见 https://devblogs.microsoft.com/dotnet/configureawait-faq

**Comment by ArgoZhang** [comment]
@SmartCode-X @BrandonStudio 昨晚发布了 9.1.3-beta02 支持了异步 Lookup 尽快使用新方法 `GetItemsByKeyAsync`。帮忙测试一下下

**Reply by densen2014** [reply]
yyds
